### DISCUSSION TITLE
Try it out button not working using root_path with ApiGateway

### DISCUSSION THREAD

**Discussion by EvanderDeLeon** [fastapi/fastapi#6096] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#main.py

from fastapi import FastAPI
from fastapi.openapi.utils import get_openapi

from test_api.router import *

logger = logging.getLogger(__name__)

app = FastAPI(
    title="test_API",
    root_path="/test",
)

app.include_router(router)

description = """
Test API overview
"""

def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema
    openapi_schema = get_openapi(
        title="Test API",
        version="0.0.1",
        description=description,
        routes=app.routes,
    )
    openapi_schema["info"]["x-logo"] = {
        "url": "https://fastapi.tiangolo.com/img/logo-margin/logo-teal.png"
    }
    app.openapi_schema = openapi_schema
    return app.openapi_schema

# assign the customized OpenAPI schema
app.openapi = custom_openapi

#router.py

import logging

from fastapi import APIRouter

logger = logging.getLogger(__name__)

router = APIRouter(
    prefix="/test",
)

@router.get(
    "/health_check",
)
def root():
    return {"status": "Success"}
```

### Description

We are using FastAPI alongside ApiGateway on AWS. We have multiple APIs being handled by one Custom domain name so we are required to use API mappings with specific Paths.

With the configuration shown all the routes for the API works as well as the documentation or getting the openapi.json file. The only issue is when trying to execute the API from the docs page with the "Try it out" option. Is not adding the prefix so every request is ending up with a 404 error. 

![image](https://user-images.githubusercontent.com/105293902/210645631-afc721e5-731a-43b6-99a6-4c71f91d2d7c.png)

We tried working with routes as well, but this ends up duplicating the prefix and we don't want that. 

After more testing we found that the issue could be related to the custom_openapi function, if we remove it, the try it out button works. But the rest of the docs does not. 

Is there any kind of parameter we could use inside this definition to achieve the addition of the prefix only to the execute part without having the need to add a router since this causes the prefix to be duplicated. 

The request would look like this: 

curl --location --request GET 'http://127.0.0.1:9999/test/test/health_check'

### Operating System

Linux

### FastAPI Version

0.88.0

### Python Version

Python 3.8

**Comment by juan-blackfynn** [comment]
Following your solution I agree that the problem could be the `custom_openapi` function inclusion and it is probably causing the issue because it seems you don't need it. You could try just deleting `custom_openapi` call (and also definition if you want) and include the API description under app -description- param. That way you will use the `root_path` the way you want to be able executing test by using try it out option. Otherwise in such the case you want to keep using `custom_openapi` you should continue extending documentation also for paths to "override" the path that doesn't include the prefix from root_path and include the one that includes the path prefix manually, just like this:


```description = """
Test API overview
"""

def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema
    openapi_schema = get_openapi(
        title="Test API",
        version="0.0.1",
        description=description,
        routes=app.routes,
    )
    openapi_schema["info"]["x-logo"] = {
        "url": "https://fastapi.tiangolo.com/img/logo-margin/logo-teal.png"
    }
    openapi_schema["paths"]["/test/health_check"] = {
        'get':
            {'summary': 'Root',
             'operationId': 'root_health_check_get',
             'responses': {
                '200': {
                    'description': 'Successful Response',
                    'content': {
                        'application/json': {
                            'schema': {}
                        }
                    }
                }
             }
        }
    }
    openapi_schema["paths"]["/health_check"] = {}
    app.openapi_schema = openapi_schema
    return app.openapi_schema

# assign the customized OpenAPI schema
app.openapi = custom_openapi```

 Hope this helps!
